src/main/java/net/staticsnow/nexus/repository/apt/internal/gpg/AptSigningHandler.java

```diff
@@ -15,37 +15,21 @@
 
 package net.staticsnow.nexus.repository.apt.internal.gpg;
 
-import static org.sonatype.nexus.repository.http.HttpMethods.GET;
-
 import javax.inject.Named;
 import javax.inject.Singleton;
 
 import org.sonatype.goodies.common.ComponentSupport;
 import org.sonatype.nexus.repository.http.HttpResponses;
-import org.sonatype.nexus.repository.view.Context;
 import org.sonatype.nexus.repository.view.Handler;
-import org.sonatype.nexus.repository.view.Response;
-
-import net.staticsnow.nexus.repository.apt.internal.snapshot.AptSnapshotHandler;
 
 @Named
 @Singleton
-public class AptSigningHandler extends ComponentSupport implements Handler {
-	@Override
-	public Response handle(Context context) throws Exception {
-		String path = assetPath(context);
-		String method = context.getRequest().getAction();
-		AptSigningFacet facet = context.getRepository().facet(AptSigningFacet.class);
-		
-		if ("repository-key.gpg".equals(path) && GET.equals(method)) {
-			return HttpResponses.ok(facet.getPublicKey());
-		}
-		
-		return context.proceed();
-	}
-	
-	private String assetPath(Context context) {
-		final AptSnapshotHandler.State snapshotState = context.getAttributes().require(AptSnapshotHandler.State.class);
-		return snapshotState.assetPath;
-	}
+public class AptSigningHandler
+    extends ComponentSupport
+{
+  public final Handler handle = context -> {
+    AptSigningFacet facet = context.getRepository().facet(AptSigningFacet.class);
+
+    return HttpResponses.ok(facet.getPublicKey());
```

Comment:
Simplified this down quite a bit. Created Handler manually (we tend to do this) instead of Implementing Handler. Put the logic from the Handler in routes and inside the Recipe, where I feel it's better placed.